### DISCUSSION TITLE
how to create my own api or how do i use existing api like in binapi etc?

### DISCUSSION THREAD

**Discussion by kolinfluence** [FDio/govpp#97] [Q&A] [answered]
1. how to create my own api or how do i use existing api like in binapi etc?
2. i saw af_xdp but i'm not sure how to even use it.

1. does anyone have any tutorial resources i can reference on using existing api?
2. what's binapi?
3. how to use af_xdp (with examples)?

**Comment by ondrej-fabry** [comment, marked as answer]
The GoVPP project provides generator of Go bindings and Go client for the VPP API. It does not in any way operate with specific API message/service related to VPP networking objects.

Please refer to the official VPP documentation: https://s3-docs.fd.io/vpp/22.10/
If that does not help you can try asking on VPP-dev mailing list: https://lists.fd.io/g/vpp-dev

To answer your questions:

1. You would need to create a VPP plugin with `*.api` definition as part of your custom VPP.
   Refer to developer guide in VPP documentation: [Adding a new plugin or feature](https://s3-docs.fd.io/vpp/22.10/developer/plugindoc/index.html#adding-a-new-plugin-or-feature)
2. Try searching 'af_xdp' in VPP documentation: [Search Results](https://s3-docs.fd.io/vpp/22.10/search.html?q=af_xdp+&check_keywords=yes&area=default)
3. Try the search in VPP documentation ^
4. _binapi_ - it's abbreviation of binary API which refers to protocol of the VPP API
   Refer to VPP documentation: https://s3-docs.fd.io/vpp/22.10/interfacing/binapi/index.html
5. See point number 2
